airflow/contrib/operators/jenkins_job_trigger_operator.py

```diff
@@ -0,0 +1,146 @@
+from airflow.utils.decorators import apply_defaults
+from airflow.contrib.hooks.jenkins_hook import JenkinsHook
+import jenkins
+import json
+from six.moves.urllib.request import Request
+from six.moves.urllib.parse import urlencode, quote_plus
+
+try:
+    basestring
+except NameError:
+    basestring = str # For python3 compatibility
+
+
+class JenkinsJobTriggerOperator(BaseOperator):
+    """
+    Trigger a Jenkins Job and monitor it's execution.
+    This operator depend on python-jenkins library, version >= 0.4.15 to communicate with jenkins server.
+    You'll also need to configure a Jenkins connection in the connections screen.
+    :param jenkins_connection_id: The jenkins connection to use for this job
+    :type jenkins_connection_id: string
+    :param job_name: The name of the job to trigger
+    :type job_name: string
+    :param parameters: The parameters block to provide to jenkins
+    :type parameters: string
+    :param sleep_time: How long will the operator sleep between each status
+    request for the job (min 1, default 10)
+    :type sleep_time: int
+    :param max_try_before_job_appears: The maximum number of requests to make
+        while waiting for the job to appears on jenkins server (default 10)
+    :type max_try_before_job_appears: int
+    """
+    template_fields = ('parameters',)
+    template_ext = ('.json',)
+    ui_color = '#f9ec86'
+
+    @apply_defaults
+    def __init__(self,
+                 jenkins_connection_id,
+                 job_name,
+                 parameters="",
+                 sleep_time=10,
+                 max_try_before_job_appears=10,
+                 *args,
+                 **kwargs):
+        super(JenkinsJobTriggerOperator, self).__init__(*args, **kwargs)
+        self.job_name = job_name
+        self.parameters = parameters
+        if sleep_time < 1:
+            sleep_time = 1
+        self.sleep_time = sleep_time
+        self.jenkins_connection_id = jenkins_connection_id
+        self.max_try_before_job_appears = max_try_before_job_appears
+
+    def build_job_with_parameters(self, jenkins_server):
+        #Workaround to avoid having a too long URL in case of long parameters
+        #(the normal build_job of python-jenkins library put all parameters in
+        #the URL and it gets too long with long parameters)
+        parametersArray = []
+        for key,val in self.parameters.items():
+            parametersArray.append({"name":key, "value":val})
+        postBody = {"parameter":parametersArray}
+        postBody = json.dumps(postBody)
+        request = Request(jenkins_server.build_job_url(self.job_name, None, None),
+            "json="+quote_plus(postBody))
+        jenkins_server.jenkins_open(request)
+
+    def get_hook(self):
+        return JenkinsHook(self.jenkins_connection_id)
+
+    def execute(self, context):
+        if not self.jenkins_connection_id:
+            self.logger.error('Please specify the jenkins connection id to use.'
```

Comment:
Please use self.log. Self.logger is deprecated

Reply:
Done :)